### PR TITLE
Update image / go and capi also calico and ccm

### PR DESCRIPTION
**Pull request by cpanato**
**What this PR does / why we need it**:

- update go to 1.17.7
- update capi to v1.1.1
- update prebuilt images
- update calico
- update ccm

pre-built images

```console
$ doctl compute image list
ID           Name                                               Type        Distribution    Slug    Public    Min Disk
63624555     cluster-api-ubuntu-1804-v1.16.2-1589476338         snapshot    Ubuntu                  false     25
80071030     Cluster API Kubernetes v1.18.16 on Ubuntu 20.04    snapshot    Ubuntu                  false     25
93100171     Cluster API Kubernetes v1.20.10 on Ubuntu 20.04    snapshot    Ubuntu                  false     25
96195649     Cluster API Kubernetes v1.22.4 on Ubuntu 20.04     snapshot    Ubuntu                  false     25
96197681     Cluster API Kubernetes v1.21.7 on Ubuntu 20.04     snapshot    Ubuntu                  false     25
101590462    Cluster API Kubernetes v1.21.9 on Ubuntu 20.04     snapshot    Ubuntu                  false     25
101591331    Cluster API Kubernetes v1.20.15 on Ubuntu 20.04    snapshot    Ubuntu                  false     25
101591787    Cluster API Kubernetes v1.22.6 on Ubuntu 20.04     snapshot    Ubuntu                  false     25
101593074    Cluster API Kubernetes v1.23.3 on Ubuntu 20.04     snapshot    Ubuntu                  false     25
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
- update go to 1.17.7
- update capi to v1.1.1
- update prebuilt images
```

### CODE REVIEW

File: test/e2e/config/digitalocean-ci.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ providers:
   - name: digitalocean
     type: InfrastructureProvider
     versions:
-    - name: v1.1.0
+    - name: v1.1.99
```
Comment (timoreimann) [inline test/e2e/config/digitalocean-ci.yaml:49]: Just to confirm, is `99` intended here?

Comment (cpanato) [reply test/e2e/config/digitalocean-ci.yaml:49]: yep, this is just to we not forget and keeping update everytime

**Comment by cpanato** [comment]
it is not ready yet, the conformance is not passing, I'm checking what is wrong

**Comment by cpanato** [comment]
now everything is working again!
/hold cancel
@timoreimann PTAL

I will add the ccm in the cluster-template in a follow up

**Pull request merged**